user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/config/CacheConfiguration.java

```diff
@@ -0,0 +1,210 @@
+                config.setInstanceName(instanceName);
+                config.getNetworkConfig().setPort(hazelcastPort); // Own port
+                for (ServiceInstance instance : discoveryClient.getInstances(serviceId)) {
+                    String clusterMember = instance.getHost() + ":" + hazelcastPort; // Address where the other instance is expected
+                    log.info("Adding Hazelcast (prod) cluster member {}", clusterMember);
+                    config.getNetworkConfig().getJoin().getTcpIpConfig().addMember(clusterMember);
+                }
+            }
+        }
+        config.getMapConfigs().put("default", initializeDefaultMapConfig(jHipsterProperties));
+        config.getMapConfigs().put("de.tum.in.www1.artemis.domain.*", initializeDomainMapConfig(jHipsterProperties));
+
+        return Hazelcast.newHazelcastInstance(config);
+    }
+
+    private void hazelcastBindOnlyOnInterface(String hazelcastInterface, Config config) {
+        // Hazelcast should bind to the interface and use it as local and public address
+        log.info("Binding Hazelcast to interface {}", hazelcastInterface);
+        System.setProperty("hazelcast.local.localAddress", hazelcastInterface);
+        System.setProperty("hazelcast.local.publicAddress", hazelcastInterface);
+        config.getNetworkConfig().getInterfaces().setEnabled(true).setInterfaces(Collections.singleton(hazelcastInterface));
+
+        // Hazelcast should only bind to the interface provided, not to any interface
+        config.setProperty("hazelcast.socket.bind.any", "false");
+        config.setProperty("hazelcast.socket.server.bind.any", "false");
+        config.setProperty("hazelcast.socket.client.bind.any", "false");
+    }
+
+    @Autowired(required = false) // ok
+    public void setGitProperties(GitProperties gitProperties) {
+        this.gitProperties = gitProperties;
+    }
+
+    @Autowired(required = false) // ok
+    public void setBuildProperties(BuildProperties buildProperties) {
+        this.buildProperties = buildProperties;
+    }
+
+    @Bean
+    public KeyGenerator keyGenerator() {
+        return new PrefixedKeyGenerator(this.gitProperties, this.buildProperties);
+    }
+
+    private MapConfig initializeDefaultMapConfig(JHipsterProperties jHipsterProperties) {
+        MapConfig mapConfig = new MapConfig();
+
+        /*
+         * Number of backups. If 1 is set as the backup-count for example, then all entries of the map will be copied to another JVM for fail-safety. Valid numbers are 0 (no
+         * backup), 1, 2, 3. While we store most of the data in the database, we might use the backup for live quiz exercises and their corresponding hazelcast hash maps
+         */
+        mapConfig.setBackupCount(jHipsterProperties.getCache().getHazelcast().getBackupCount());
+
+        /*
+         * Valid values are: NONE (no eviction), LRU (Least Recently Used), LFU (Least Frequently Used). LRU is the default.
```

Comment:
I think you refer here to the eviction policy? Even though this is rather obvious if one takes a look at the code, it makes sense to specify this in the documentation so the reader doesn't have to search first.